### PR TITLE
openstack: Create/Remove cinder volume snapshot

### PR DESCRIPTION
**Pull request by nhicher**
##### SUMMARY
Add a module to allow user to create or remove cinder block storage volume snapshot. This module is based on os_volume.py module

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
lib/ansible/modules/cloud/openstack/os_volume_snapshot.py

##### ANSIBLE VERSION
```
ansible 2.3.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Jun 26 2017, 10:20:05) [GCC 7.1.1 20170622 (Red Hat 7.1.1-3)]
```

### CODE REVIEW

**Comment by nhicher** [comment]
Any chance to have a review ?

**Comment by ghost** [comment]
The code here looks fine, this needs test support though

**Comment by nhicher** [comment]
Hello @SamYaple  thanks for the review
Do you have an example about the tests please ? I found a test for test/units/modules/cloud/openstack/test_os_server.py, but nothing for the others modules.

**Comment by ghost** [comment]
https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/openstack/os_keystone_service.py#L181

File: lib/ansible/modules/cloud/openstack/os_volume_snapshot.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+    module = AnsibleModule(argument_spec=argument_spec)
+
+    if not HAS_SHADE:
+        module.fail_json(msg='shade is required for this module')
+
+    state = module.params['state']
+
+    if module.check_mode:
+        module.exit_json(changed=_system_state_change(module, service))
```
Comment (ghost) [inline lib/ansible/modules/cloud/openstack/os_volume_snapshot.py:167]: you have no function named '_system_state_change'. You need to work that in, see theother module as an example.

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by ruizink** [comment]
This feature is already provided by #19122

**Comment by mnaser** [comment]
Can we close this?

**Comment by mnaser** [comment]
close_me

**Pull request closed without merging**